native/mod_proxy_cluster/mod_proxy_cluster.c

```diff
@@ -1510,17 +1562,40 @@ static void update_workers_lbstatus(proxy_server_conf *conf, apr_pool_t *pool, s
                 apr_status_t rv;
                 apr_pool_t *rrp;
                 request_rec *rnew;
-                proxy_worker *worker = get_worker_from_id_stat(conf, id[i], stat);
+                proxy_worker *worker;
+                apr_thread_mutex_lock(lock);
+                worker = get_worker_from_id_stat(conf, id[i], stat, ou);
+                apr_thread_mutex_unlock(lock);
+
                 if (worker == NULL)
                     continue; /* skip it */
 #if AP_MODULE_MAGIC_AT_LEAST(20101223,1)
                 apr_snprintf(sport, sizeof(sport), ":%d", worker->s->port);
+
+                if (strcmp(worker->s->scheme, ou->mess.Type) ||
+                    strcmp(worker->s->hostname, ou->mess.Host) ||
+                    strcmp(sport, ou->mess.Port)) {
+                    /* the worker doesn't correspond to the node */
+                    worker->s->index = 0;
+                    /* XXX: broken ap_my_generation--; mark old generation that will recreate the process */ 
```

Comment:
There is no way for the moment to force the child process to stop in a clean way,  ap_my_generation-- doesn't work (nor compile).